src/app/nhs-app-template-submitted/[templateId]/page.tsx

```diff
@@ -1,11 +1,21 @@
 'use server';
 
-const NhsAppTemplateSubmittedPage = async () => {
-  return (
-    <h2 className='nhsuk-heading-l' data-testid='page-sub-heading'>
-      Placeholder page
-    </h2>
-  );
+import { TemplateSubmitted } from '@molecules/TemplateSubmitted/TemplateSubmitted';
+import { TemplateSubmittedPageProps } from '@utils/types';
+import { getTemplate } from '@utils/form-actions';
+import { redirect, RedirectType } from 'next/navigation';
+
+const PreviewNhsAppTemplatePage = async ({
+  params: { templateId },
+}: TemplateSubmittedPageProps) => {
+  const template = await getTemplate(templateId);
+  if (!template) {
+    redirect('/invalid-template', RedirectType.replace);
+  }
+
+  const { name } = template;
+
+  return <TemplateSubmitted templateId={templateId} templateName={name} />;
```

Comment:
Might be better to use the if from the database rather than what is passed in from params.

Reply:
More that it establishes a safer pattern of transforming request data into validated data which is then subsequently used.  Possibly an argument to refactor the last bit into a function that takes a template as its input to keep the raw user data out of scope.  (Although this is all very subjective and "nit"-y as folk like to say).

Reply:
updated